server/libs/common/src/services/user.service.ts

```diff
@@ -0,0 +1,153 @@
+import { BadRequestException, ForbiddenException, Inject, Injectable, UnauthorizedException } from '@nestjs/common';
+import * as bcrypt from 'bcrypt';
+import { User } from '../models';
+
+export type UserCreateDto = Pick<User, 'email'> & Partial<User>;
```

Comment:
Are we going to move these DTOs into their respective directory? Is your plan going to discard the existing DTOs in all api modules?

Reply:
We probably can replace most of the api dtos with these in the future, but that would impact the types in the open api libraries, leading to significant work in the web/mobile apps.

Whether these are export from the service or moved to a separate folder/file, I don't have strong opinions about that.

Reply:
With the approach or having DTOs as a class, you can add decorator for each property, can you do that  with this approach?